packages/taucmdr/model/experiment.py

```diff
@@ -155,13 +143,13 @@ def create(self, data):
         return super(ExperimentController, self).create(data)
 
     def update(self, data, keys):
-        return super(ExperimentController, self).update(data, self._restrict_project(keys))
+        return super(ExperimentController, self).update(data, keys)
 
     def unset(self, fields, keys):
-        return super(ExperimentController, self).unset(fields, self._restrict_project(keys))
+        return super(ExperimentController, self).unset(fields, keys)
 
     def delete(self, keys):
-        return super(ExperimentController, self).delete(self._restrict_project(keys))
+        return super(ExperimentController, self).delete(keys)
```

Comment:
I'm not sure that removing the `_restrict_project` calls everywhere is appropriate, since this allows these commands to work on objects in projects other than the currently selected one. In `unstable` an experiment in one project can't be deleted when another project is selected:

```
% tau init
[...]
== Experiments in project 'mm' =======================================================================================================================================================================================================================================================================================================================================================================

+------------------+--------+-----------+--------+-------------+-------------+---------------+-----------------------+
|       Name       | Trials | Data Size | Target | Application | Measurement | Record Output |     TAU Makefile      |
+==================+========+===========+========+=============+=============+===============+=======================+
| delphi-mm-sample |   0    |   0.0B    | delphi |     mm      |   sample    |       0       | Makefile.tau-a7309670 |
+------------------+--------+-----------+--------+-------------+-------------+---------------+-----------------------+
[...]
% tau project create foo
[TAU] Created a new project named 'foo'.
% tau project select foo
% tau experiment delete delphi-mm-sample
[...]
tau experiment delete: error: No project-level experiment with name='delphi-mm-sample'.
```

If this change is applied, then I can delete `delphi-mm-sample` even though it's in a different project:

```
tau experiment delete delphi-mm-sample
[TAU] Deleted experiment 'delphi-mm-sample'
```

I think this isn't the desired behavior. (Although I would note that `_restrict_project` doesn't exist for the other data types, so TAU Commander will happily let you delete a Target, Application, or Measurement from a different project, and in fact won't let you reuse a name for one of those objects in multiple projects. I think that behavior is also wrong...)

Reply:
This is now obsolete with the changes to the controllers.